selftest.py

```diff
@@ -179,9 +178,9 @@ def prior_test_results(cls, _db, constructor_method=None, *args, **kwargs):
         constructor_method = constructor_method or cls
         integration = None
         instance = constructor_method(*args, **kwargs)
-        integration = instance.external_integration(_db)
+        integration = instance.external_integration(_db) or instance.search_integration(_db)
```

Comment:
This is another argument for eventually creating subclasses of HasSelfTests that can remove this case-by-case logic from the core HasSelfTests code.

For now, I think it's better to use `isinstance` to be explicit about the classes we are expecting. I think this will work:

```
if isinstance(instance, ExternalSearchIndex):
    integration = instance.search_integration(_db)
else:
    integration = instance.external_integration(_db)
```